File: fx-backend/src/main/java/com/fxproject/fxbackend/repository/FxOperacion.java
package com.fxproject.fxbackend.repository;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.Data;

import java.time.Instant;
import java.time.LocalDate;

@Entity
@Data
public class FxOperacion {
    @Id
    private int id;
    private String monedaOrigen;
    private String monedaDestino;
    private double montoOrigen;
    @GeneratedValue
    @Column(insertable = false, updatable = false)
    private double montoDestino;
    private double tipoCambio;
    private String estado;
    private LocalDate fechaOperacion;
    private Instant createdAt;
}

File: fx-frontend/src/app/model/fx-operacion.ts
export interface FxOperacion {
 id: number,
 monedaOrigen: string,
 monedaDestino: string,
 montoOrigen: number,
 montoDestino: number;
 tipoCambio: number,
 estado: string;
 fechaOperacion: Date,
 createdAt: Date;
}